backend/services/implementations/testSessionService.ts

```diff
@@ -163,13 +186,42 @@ class TestSessionService implements ITestSessionService {
     limit?: number,
   ): Promise<Array<TestSessionResponseDTO>> {
     try {
-      const query = MgTestSession.find({ teacher: { $eq: teacherId } });
+      const now = new Date();
+
+      let testSessions: TestSession[];
       if (limit !== undefined) {
-        query.limit(limit);
+        const queries = [
+          // active
+          MgTestSession.find({
+            teacher: { $eq: teacherId },
+            endDate: { $gte: now },
+            startDate: { $lte: now },
+          }).sort({ endDate: 1 }),
+          // upcoming
+          MgTestSession.find({
+            teacher: { $eq: teacherId },
+            startDate: { $gt: now },
+          }).sort({ startDate: 1 }),
+          // past
+          MgTestSession.find({
+            teacher: { $eq: teacherId },
+            endDate: { $lt: now },
+          }).sort({ startDate: -1 }),
```

Comment:
should this be endDate: -1? then you can see the most recently completed ones

Reply:
Unfortunately the UI shows past assessments sorted by "assigned" date rather than "closed" date (I believe this was a design decision). We could override that but I'd have to change the frontend copy. For now it has to be startDate